WordPressKit/JSONDecoderExtension.swift

```diff
@@ -0,0 +1,47 @@
+import Foundation
+
+extension JSONDecoder {
+
+    static var apiDecoder: JSONDecoder {
+        let decoder = JSONDecoder()
+        decoder.dateDecodingStrategy = JSONDecoder.DateDecodingStrategy.supportMultipleDateFormats
+        decoder.keyDecodingStrategy = .convertFromSnakeCase
+        return decoder
+    }
+}
+
+extension JSONDecoder.DateDecodingStrategy {
+    
+    enum DateFormat: String {
+        case noTime = "yyyy-mm-dd"
+        case dateWithTime = "yyyy-MM-dd HH:mm:ss"
+        
+        var formatter: DateFormatter {
+            let dateFormatter = DateFormatter()
+            dateFormatter.dateFormat = rawValue
+            return dateFormatter
+        }
+    }
+    
+    static var supportMultipleDateFormats: JSONDecoder.DateDecodingStrategy {
+        return JSONDecoder.DateDecodingStrategy.custom({ (decoder) -> Date in
+            let container = try decoder.singleValueContainer()
+            let dateStr = try container.decode(String.self)
+            
+            let len = dateStr.count
+            var date: Date? = nil
+            if len == DateFormat.noTime.rawValue.count {
+                date = DateFormat.noTime.formatter.date(from: dateStr)
+            } else if len == DateFormat.dateWithTime.rawValue.count {
+                date = DateFormat.dateWithTime.formatter.date(from: dateStr)
+            }
+            guard let _date = date else {
```

Comment:
stylistic nitpick: I don't like the use of underscore here — I think it's fairly incosistent with the rest of the codebase.

I think this would read better like this:
```swift
let date: Date

switch len {
case DateFormat.noTime.rawValue.count:
    date = DateFormat.noTime.formatter.date(from: dateStr)
case DateFormat.dateWithTime.rawValue.count:
    date = DateFormat.dateWithTime.formatter.date(from: dateStr)
default:
    throw DecodingError.dataCorruptedError(
                     in: container,
                     debugDescription: "Cannot decode date string \(dateStr)"
                 )
}
```

That way you get to avoid both optionals and the mutability here :) (and I think it's slightly more clear what the code is doing, but that's a personal thing)

Reply:
oh I like what you suggested here, the thing is DateFormat.noTime.formatter.date(from: dateStr) also returns Optional result so I could only partially apply your solution. In the new commit I got rid of the underscore also.